cdap-client/src/main/java/co/cask/cdap/client/config/ConnectionConfig.java

```diff
@@ -66,10 +68,15 @@ public ConnectionConfig(Id.Namespace namespace, String hostname, int port, boole
     this.hostname = hostname;
     this.port = port;
     this.sslEnabled = sslEnabled;
+    if (SecurityRequestContext.getUserId().isPresent()) {
```

Comment:
Have you tried this out? This will never be present in the CLI, as we don't set it. It's only set in CDAP Master (server-side). To get the username, you'll need to capture it when the user provides username/password upon connecting to a secure CDAP instance (see `ConnectCommand.execute()` which calls `CLIConfig.tryConnect`).